File: exts/jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/UXNumberSpinner.java
package org.develnext.jphp.ext.javafx.classes;

import javafx.geometry.Pos;
import javafx.scene.control.SpinnerValueFactory;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import org.develnext.jphp.ext.javafx.support.control.NumberSpinner;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Getter;
import php.runtime.annotation.Reflection.Property;
import php.runtime.annotation.Reflection.Setter;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Reflection.Name(JavaFXExtension.NS + "UXNumberSpinner")
public class UXNumberSpinner extends UXSpinner<NumberSpinner> {
    interface WrappedInterface {
        @Property int min();
        @Property int max();
        @Property int step();
        @Property int initial();
        @Property NumberSpinner.ArrowsStyle arrowsStyle();
    }

    public UXNumberSpinner(Environment env, NumberSpinner wrappedObject) {
        super(env, wrappedObject);
    }

    public UXNumberSpinner(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Override
    public void __construct() {
        __wrappedObject = new NumberSpinner();
    }

    @Getter
    public int getValue() {
        return getWrappedObject().getValue();
    }

    @Setter
    public void setValue(int value) {
        getWrappedObject().getValueFactory().setValue(value);
    }

    @Setter
    public void setAlignment(Pos value) {
        getWrappedObject().getEditor().setAlignment(value);
    }

    @Getter
    public Pos getAlignment() {
        return getWrappedObject().getEditor().getAlignment();
    }
}

File: exts/jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/UXNumberSpinner.php
<?php
namespace php\gui;

/**
 * @package php\gui
 * @packages gui, javafx
 */
class UXNumberSpinner extends UXSpinner
{
    /**
     * @var int
     */
    public $min = 0x80000000;

    /**
     * @var int
     */
    public $max = 0x7fffffff;

    /**
     * @var int
     */
    public $step = 1;

    /**
     * @var int
     */
    public $initial = 0;

    /**
     * @var int
     */
    public $value = 0;

    /**
     * RIGHT_VERTICAL, RIGHT_HORIZONTAL, LEFT_VERTICAL, LEFT_HORIZONTAL, SPLIT_VERTICAL, SPLIT_HORIZONTAL
     *
     * @var string
     */
    public $arrowsStyle = 'RIGHT_VERTICAL';
}